infra-examples/aws/k8s-cluster/main.tf

```diff
@@ -0,0 +1,220 @@
+      type                       = "ingress"
+      source_node_security_group = true
+    }
+    egress_all = {
+      description      = "openedx-k8s-harmony: Node all egress"
+      protocol         = "-1"
+      from_port        = 0
+      to_port          = 0
+      type             = "egress"
+      cidr_blocks      = ["0.0.0.0/0"]
+      ipv6_cidr_blocks = ["::/0"]
+    }
+  }
+
+  eks_managed_node_groups = {
+    # This node group is managed by Karpenter. There must be at least one
+    # node in this group at all times in order for Karpenter to monitor
+    # load and act on metrics data. Karpenter's bin packing algorithms
+    # perform more effectively with larger instance types. The default
+    # instance type is t3.large (2 vCPU / 8 GiB). These instances,
+    # beyond the 1 permanent instance, are assumed to be short-lived
+    # (a few hours or less) as these are usually only instantiated during
+    # bursts of user activity such as at the start of a scheduled lecture or
+    # exam on a large mooc.
+    service = {
+      capacity_type     = "SPOT"
+      enable_monitoring = false
+      desired_size      = var.eks_service_group_desired_size
+      min_size          = var.eks_service_group_min_size
+      max_size          = var.eks_service_group_max_size
+
+      # for node affinity
+      labels = {
+        node-group = "service"
+      }
+
+      iam_role_additional_policies = {
+        # Required by Karpenter
+        AmazonSSMManagedInstanceCore = "arn:${local.partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"
+
+        # Required by EBS CSI Add-on
+        AmazonEBSCSIDriverPolicy = data.aws_iam_policy.AmazonEBSCSIDriverPolicy.arn
+      }
+
+      instance_types = ["${var.eks_service_group_instance_type}"]
+      tags = merge(
+        local.tags,
+        { Name = "eks-${var.shared_resource_identifier}" }
+      )
+    }
+
+  }
+}
+
+#------------------------------------------------------------------------------
+#                           SUPPORTING RESOURCES
+#------------------------------------------------------------------------------
+data "template_file" "eks-console-full-access" {
+  template = file("${path.module}/yml/eks-console-full-access.yaml")
+  vars     = {}
+}
+
+# add an AWS IAM Role definition providing AWS console access to 
+# AWS EKS cluster instances.
+resource "kubectl_manifest" "eks-console-full-access" {
+  yaml_body = data.template_file.eks-console-full-access.rendered
+}
+
+# to enable shell access to nodes from kubectl
+resource "aws_security_group" "worker_group_mgmt" {
+  name_prefix = "${var.name}-eks_hosting_group_mgmt"
+  description = "openedx-k8s-harmony: Ingress CLB worker group management"
+  vpc_id      = data.aws_vpc.reference.id
+
+  ingress {
+    description = "openedx-k8s-harmony: Ingress CLB"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+
+    cidr_blocks = [
+      "10.0.0.0/8",
+    ]
+  }
+
+  tags = local.tags
+
```

Comment:
```suggestion
```

Reply:
i'm pleased to report that this blank line has been removed :)